src/test/java/org/jabref/logic/texparser/TexParserTest.java

```diff
@@ -0,0 +1,294 @@
+package org.jabref.logic.texparser;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.fileformat.BibtexParser;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.texparser.Citation;
+import org.jabref.model.texparser.CrossingKeysResult;
+import org.jabref.model.texparser.TexParserResult;
+import org.jabref.model.util.DummyFileUpdateMonitor;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+class TexParserTest {
+    private final String DARWIN = "Darwin1888";
+    private final String EINSTEIN = "Einstein1920";
+    private final String UNRESOLVED = "UnresolvedKey";
+    private ImportFormatPreferences importFormatPreferences;
+
+    @BeforeEach
+    void setUp() {
+        importFormatPreferences = mock(ImportFormatPreferences.class, Answers.RETURNS_DEEP_STUBS);
+    }
+
+    @AfterEach
+    void tearDown() {
+        importFormatPreferences = null;
+    }
+
+    private void testCite(String key, String citeString, boolean match) {
```

Comment:
Personally, I would inline this method in `testMatchCite` and `testNonMatchCite` since the code duplication is essentially only 2 lines.